src/manifolds/StiefelEuclideanMetric.jl

```diff
@@ -110,6 +110,22 @@ end
 
 _euclidean_unit_vector(n, i) = [k == i ? 1.0 : 0.0 for k in 1:n]
 
+"""
+    inverse_retract(M::Stiefel, p, q, method::ProjectionInverseRetraction)
+
+Compute a projection-based inverse retraction.
+
+The retraction is computed by projecting the logarithm map in the embedding to the tangent
+space at ``p``.
+"""
+inverse_retract(::Stiefel, ::Any, ::Any, ::ProjectionInverseRetraction)
+
+function inverse_retract_project!(M::Stiefel, X, p, q)
+    log!(get_embedding(M), X, p, q)
```

Comment:
Maybe this should be using `embed` on the points before calling `log!`?

Reply:
Why can't it just do `X .= p .- q`?

Reply:
It could be; indeed that's what it evaluates to. But I'm trying to be explicit about how we're performing the inverse retraction in the embedding and then projecting. Seems more general, especially if we define custom points.

Reply:
Since we are on Stiefel we could really just use `X .= p .- q` here.

For special Stiefel point representations I would expect them to overload this?
Otherwise, for the very general implementation you aim for, you could also fall back to implementing this generically for embedded manifolds?

Reply:
> Since we are on Stiefel we could really just use `X .= p .- q` here.

For simplicity, I went with this approach. A generic default for embedded manifolds could be implemented in a separate PR.

Reply:
Yes, I think a generic implementation would be nice as well :) For here the simple approach should be enough :)